internal/collectors/gitlab/repository_collector.go

```diff
@@ -0,0 +1,189 @@
+package gitlab
+
+import (
+	"github.com/Legit-Labs/legitify/internal/clients/gitlab"
+	"github.com/Legit-Labs/legitify/internal/collected/gitlab_collected"
+	"github.com/Legit-Labs/legitify/internal/collectors"
+	"github.com/Legit-Labs/legitify/internal/common/group_waiter"
+	"github.com/Legit-Labs/legitify/internal/common/permissions"
+	"github.com/Legit-Labs/legitify/internal/common/types"
+	"github.com/Legit-Labs/legitify/internal/context_utils"
+
+	gitlab2 "github.com/xanzy/go-gitlab"
+	"log"
+
+	"github.com/Legit-Labs/legitify/internal/common/namespace"
+	"golang.org/x/net/context"
+)
+
+type repositoryCollector struct {
+	collectors.BaseCollector
+	Client  *gitlab.Client
+	Context context.Context
+}
+
+func NewRepositoryCollector(ctx context.Context, client *gitlab.Client) collectors.Collector {
+	c := &repositoryCollector{
+		Client:  client,
+		Context: ctx,
+	}
+	collectors.InitBaseCollector(&c.BaseCollector, c)
+	return c
+}
+
+func (rc *repositoryCollector) Namespace() namespace.Namespace {
+	return namespace.Repository
+}
+
+func (rc *repositoryCollector) CollectMetadata() collectors.Metadata {
+	res := collectors.Metadata{}
+
+	repositories, exist := context_utils.GetRepositories(rc.Context)
+
+	if exist {
+		for _, repository := range repositories {
+			_, _, err := rc.Client.Client().Projects.GetProject(repository, &gitlab2.GetProjectOptions{})
```

Comment:
You can remove this check since we check it when we process the command line arguments